Terminal.Gui/Views/SpinnerView/SpinnerStyle.cs

```diff
@@ -0,0 +1,1699 @@
+﻿//------------------------------------------------------------------------------
+// SpinnerStyles below are derived from
+// <https://github.com/sindresorhus/cli-spinners/blob/master/spinners.json>
+// MIT License
+// Copyright (c) Sindre Sorhus <sindresorhus@gmail.com>
+// (https://sindresorhus.com)
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//------------------------------------------------------------------------------
+// Windows Terminal supports Unicode and Emoji characters, but by default
+// conhost shells (e.g., PowerShell and cmd.exe) do not. See
+// <https://spectreconsole.net/best-practices>.
+//------------------------------------------------------------------------------
+
+using System;
+
+namespace Terminal.Gui {
+	/// <summary>
+	/// SpinnerStyles used in a <see cref="SpinnerView"/>.
+	/// </summary>
+	public abstract class SpinnerStyle {
+		const int DEFAULT_DELAY = 80;
+		const bool DEFAULT_BOUNCE = false;
+		const bool DEFAULT_SPECIAL = false;
+
+		/// <summary>
+		/// Gets or sets the number of milliseconds to wait between characters
+		/// in the spin.  Defaults to the SpinnerStyle's Interval value.
+		/// </summary>
+		/// <remarks>This is the maximum speed the spinner will rotate at.  You still need to
+		/// call <see cref="View.SetNeedsDisplay()"/> or <see cref="SpinnerView.AutoSpin"/> to
+		/// advance/start animation.</remarks>
+		public abstract int SpinDelayInMilliseconds { get; }
```

Comment:
IMO, delays & timers etc... are almost always in ms. I'd prefer this just be `SpinDelay`; tersness is always better. 

Of course the API docs should call that out too.

Reply:
We could go even clearer and just use a `TimeSpan` type e.g.

```csharp
/// <summary>
/// Gets or sets the time to wait between characters in the spin.  Defaults to 250 milliseconds.
/// </summary>
/// <remarks>This is the maximum speed the spinner will rotate at.  You still need to
/// call <see cref="View.SetNeedsDisplay()"/> or <see cref="SpinnerView.AutoSpin"/> to
/// advance/start animation.</remarks>
public TimeSpan SpinDelay { get; set; } = TimeSpan.FromMilliseconds(250);
```